### PR TITLE
refactor(rust): Add `FromIterator` impls for `PlSmallStr`

### PR DESCRIPTION
**Pull request by nameexhaustion**
From `PlSmallStr`, `&PlSmallStr`, and all the iterator types that `String::from_iter` supports

### CODE REVIEW

**Comment by ritchie46** [comment]
I am wondering if we should. As a `collect` now "seems" to work, but it goes via a `String` allocation, which might not be that clear to the callee.

**Comment by orlp** [comment]
I mean it should be easy enough to swap the impl once we use `compact_str` (my suggestion for the backing crate for now).

**Comment by nameexhaustion** [comment]
> I mean it should be easy enough to swap the impl once we use `compact_str` (my suggestion for the backing crate for now).

Ah, yes - we indeed now have native `From` impls provided by `compact_str` - have pushed a commit to update

File: crates/polars-utils/src/pl_str.rs [hunk 1]
```diff
@@ -125,7 +125,72 @@ impl From<&String> for PlSmallStr {
     }
 }
 
-/// FromIterator impls (TODO)
+impl From<Inner> for PlSmallStr {
```
Comment (nameexhaustion) [inline crates/polars-utils/src/pl_str.rs:128]: drive-by - add `From<Inner>`

**Review by ritchie46 — APPROVED** [review]
